CrossPare/src/main/java/de/ugoe/cs/cpdp/execution/CrossValidationExperiment.java

```diff
@@ -188,29 +189,29 @@ public void run() {
                     LOGGER.info(String.format("[%s] [%02d/%02d] %s: applying setwise preprocessor %s",
                                 this.config.getExperimentName(), versionCount, testVersionCount,
                                 testVersion.getVersion(), processor.getClass().getName()));
-                    processor.apply(testdata, traindataSet);
+                    processor.apply(testversion, trainversionSet);
                 }
                 for (ISetWiseProcessingStrategy processor : this.config
                     .getSetWisePostprocessors())
                 {
                 	LOGGER.info(String.format("[%s] [%02d/%02d] %s: applying setwise postprocessor %s",
                                 this.config.getExperimentName(), versionCount, testVersionCount,
                                 testVersion.getVersion(), processor.getClass().getName()));
-                    processor.apply(testdata, traindataSet);
+                    processor.apply(testversion, trainversionSet);
                 }
-                Instances traindata = makeSingleTrainingSet(traindataSet);
+                SoftwareVersion trainversion = AbstractCrossProjectExperiment.makeSingleVersionSet(trainversionSet);

```

Comment:
Why do you need to specifically write 
`AbstractCrossProjectExperiment.makeSingleVersionSet(trainversionSet);` 
instead of just 
`makeSingleVersionSet(trainversionSet);`

Reply:
CrossValidationExperiment does not extend AbstractCrossProjectExperiment and does not implement makeSingleVersionSet(). Do you want me to change that?

Reply:
Ah, that makes sense. No, it would be better to move makeSingleVersionSet in that case. I would suggest to move this to `de.ugoe.cs.cpdp.util.CrosspareUtils`